Bestride_Logic.lua

```diff
@@ -404,7 +404,14 @@ function Bestride:Swimming(flyable, zone, broom)
 	local aquaticForm = 783
 
 	local timer, _,_,scale,_,_ = GetMirrorTimerInfo(2)
-	local name, _, _, _, _, _, _, _, _, _, _ = UnitBuff("player", BestrideLocale.Misc.WaterBreathing)
+	local i;
+	for i=1,40 do
+		local name, _, _, _, _, _, _, _, _, _, _ = UnitBuff("player", i)
+		if (name == BestrideLocale.Misc.WaterBreathing or name == Bestride:SpellToName(5697)) then
```

Comment:
Doesn't do "noting", exactly. It prevents calling a mount if "Water Breathing" or Unending Breath are active.

Only reason I could think of it having been there in the first place is that most water breathing buffs also have a swim speed boost inherent to them which is in some cases faster than a mount. So, that's what I went off.

Reply:
The code didn't used to return before, that's why I thought it wasn't being used. Your logic about the movement speed boost does make sense though. Dudes the lock buff do the same?

Reply:
Lock buff is +20% swim speed.

Reply:
Ok, sounds good. Only change I would recommend then is to just "return" rather than return the name to match the "return"s in the rest of the function.

Thanks for being so thorough